pipeline-runner/R/handle_data.R

```diff
@@ -250,6 +250,19 @@ upload_matrix_to_s3 <- function(pipeline_config, experiment_id, data) {
   return(object_key)
 }
 
+upload_debug_folder_to_s3 <- function(debug_subdir, pipeline_config) {
+  fnames <- list.files(file.path("/debug", debug_subdir))
+
+  message("Uploading logs and dump file to S3 bucket ", pipeline_config$debug_bucket, " with prefix ", debug_subdir, "...")
```

Comment:
`debug_subdir` is actually experiment id + the timestamp. I see that you generate the timestamp in `init.R`, but the  timestamp actually belongs in this function and it is used only here. I suggest that:
- you remove the timestamp generation from `init.R` and add it inside this function
- remove this `debug_subdir` in `init.R` and only pass in `input$experimentId` to `upload_debug_folder_to_s3`
- in `handle_data.R`, rename the parameter called `debug_subdir` to `experimentId`.
- in `handle_data.R` generate the debug_subdir (but better call it `s3_path`) by concatenating `experimentId` with the calculated timestamp, adding a `/` between them

Reply:
I see. In this case, I would move the logic of generating the s3 prefix string inside the `upload_debug_folder_to_s3 ` function. To the `upload_debug_folder_to_s3`, I will just pass the timestamp and the experiment id

Reply:
The benefit is just that the prefix is already generated in init.R as it is used there seperately. I will duplicate it inside `upload_debug_folder_to_s3` if you would like

Reply:
I didn't realise that it's used in init.R. When I looked, I didn't see the usage. In this case, better to keep the generation only in init.R